crates/iceberg-catalog/src/api/management/v1/warehouse.rs

```diff
@@ -26,29 +30,101 @@ pub struct CreateWarehouseResponse {
     pub warehouse_id: uuid::Uuid,
 }
 
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+#[serde(rename_all = "kebab-case")]
+pub struct UpdateWarehouseStorageRequest {
+    /// Storage profile to use for the warehouse.
+    pub storage_profile: StorageProfile,
+    /// Optional storage credential to use for the warehouse.
+    /// The existing credential is not re-used. If no credential is
+    /// provided, we assume that this storage does not require credentials.
+    pub storage_credential: Option<StorageCredential>,
+}
+
+#[derive(Debug, Deserialize, ToSchema, utoipa::IntoParams)]
+#[serde(rename_all = "kebab-case")]
+pub struct ListWarehouseRequest {
+    /// Optional filter to include inactive warehouses.
+    #[serde(default)]
+    pub include_inactive: Option<bool>,
+    /// The project ID to list warehouses for.
+    /// Setting a warehouse is required.
+    #[serde(default)]
+    pub project_id: Option<uuid::Uuid>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+#[serde(rename_all = "kebab-case")]
+pub struct RenameWarehouseRequest {
+    /// New name for the warehouse.
+    pub new_name: String,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+#[serde(rename_all = "kebab-case")]
+pub struct ListProjectsResponse {
+    /// List of project IDs.
+    pub project_ids: Vec<uuid::Uuid>,
```

Comment:
Let's make this a `Vec<Project>` with  `struct Project {id: Uuid}`, I'd imagine, we'd later want to extend `ListProjectsResponse` by a human readable project name / label or other fields, if we use a struct here we can make these changes non-breaking.

Reply:
Done